validation/analyze_qam.py

```diff
@@ -0,0 +1,152 @@
+        total_mean_esa = []
+        total_mean_nersc = []
+    else:
+        rows.append(['No.', 'Scene ID'])
+
+    for i, sar_id in enumerate(d_esa):
+        mean_esa = d_esa[sar_id]['Mean'][0]
+        mean_nersc = d_nersc[sar_id]['Mean'][0]
+        total_mean_esa.append(mean_esa)
+        total_mean_nersc.append(mean_nersc)
+
+        if print_RQM:
+            rows_RQM.append([i, sar_id, '%.3f' % mean_esa,
+                         '%.3f' % mean_nersc])
+
+        rows.append([i, sar_id])
+
+    if print_RQM:
+        rows_RQM.append(['Total', '', '%.3f' % np.nanmean(total_mean_esa),
+                     '%.3f' % np.nanmean(total_mean_nersc)])
+        rows_RQM.append(['STD', '', '%.3f' % np.nanstd(total_mean_esa),
+                     '%.3f' % np.nanstd(total_mean_nersc)])
+
+    if print_RQM:
+        table_RQM = Texttable()
+        table_RQM.set_cols_align(["c"] * 4)
+        table_RQM.set_deco(Texttable.HEADER | Texttable.VLINES)
+        table_RQM.add_rows(rows_RQM)
+
+    table = Texttable()
+    table.set_cols_align(["c"] * 2)
+    table.set_deco(Texttable.HEADER | Texttable.VLINES)
+    table.add_rows(rows)
+
+    # Save table to file
+    if print_RQM:
+        with open('%s/%s_table_RQM.txt' % (out_path, data_name), 'w') as f:
+            f.write(tabulate(rows_RQM, headers='firstrow', tablefmt='latex'))
+
+    with open('%s/%s_table.txt' % (out_path, data_name), 'w') as f:
+        f.write(tabulate(rows, headers='firstrow', tablefmt='latex'))
+
+    # Calculate mean difference
+    mean_diff = np.array(total_mean_esa) - np.array(total_mean_nersc)
+
+    return np.nanmean(total_mean_esa), np.nanmean(total_mean_nersc),\
+           np.nanstd(total_mean_esa), np.nanstd(total_mean_nersc), np.nanmean(mean_diff), np.nanstd(mean_diff)
+
+def plot_results(mean_esa, mean_nersc, std_esa, std_nersc, mean_diff, mean_diff_std, out_path, data_name):
+    labels = ['ESA', 'NERSC', 'Mean difference\n(ESA-NERSC)']
+    plt.clf()
+    title = '%s RQM' % data_name
+
+    # Create lists for the plot
+    x_pos = np.arange(len(labels))
+    data = [mean_esa, mean_nersc, mean_diff]
+    error = [std_esa, std_nersc, mean_diff_std]
+
+    # Build the plot
+    fig, ax = plt.subplots()
+    ax.bar(x_pos, data, yerr=error, align='center', alpha=0.5,
+           color='#82B446', ecolor='black', capsize=10)
+    ax.set_ylabel('RQM')
+    ax.set_xticks(x_pos)
+    ax.set_xticklabels(labels)
+    ax.set_title(title)
+    ax.yaxis.grid(True)
+
+    # Save the figure and show
+    plt.tight_layout()
+    plt.savefig('%s/%s_RQM.png' % (out_path, data_name), bbox_inches='tight', dpi=300)
+
+args = parse_run_experiment_args()
+
+ffiles_mask_esa = os.path.join(args.json_path, '*ESA*RQM*.json')
+ffiles_esa = sorted(glob.glob(ffiles_mask_esa))
+
+ffiles_mask_nersc = os.path.join(args.json_path, '*NERSC*RQM*.json')
+ffiles_nersc = sorted(glob.glob(ffiles_mask_nersc))
+
+for esa_out_fname in ffiles_esa:
+    for nersc_out_fname in ffiles_nersc:
```

Comment:
Nested loops?
Do you compare all nersc files with all esa files? 
Apparently you hvae different regions in differenmt subdirs, so your *ESA*RQM*.json contains rqm values from ocean, arctic, deserts, etc. Right? And the same for NERSC. So in the nested loops you compare all combinations:
esa_arctic with nersc_arctic
esa_arctic with nersc_ocean
esa_arctic with nersc_desert
esa_ocean with nersc_arctic
esa_ocean with nersc_ocean
esa_ocean with nersc_desert
esa_arctic with nersc_arctic
and so on...

Why don't you use zip()?